### ISSUE TITLE
Tests rely on hardcoded /tmp/diskcache directory

### ISSUE THREAD

**Issue by mgorny** [grantjenks/python-diskcache#145]
A unique temporary directory (aka mkdtemp) should be used instead. The problems with hardcoding the path are:

1. It assumes the path isn't used by user (otherwise you're going to remove some real data!).
2. It assumes only one instance of test suite is run ever.
3. Since the test directory is not cleaned up properly, it assumes no other user ever run the test suite before.

And yes, I've discovered this precisely because I wasn't able to run tests because `/tmp/diskcache` already existed and was owned by another user.

**Comment by grantjenks** [comment]
That’s fair. I think some tests take advantage of the single named directory to validate specific behaviors. It may not be so simple to fix in all cases. It also may be a simple fix and I just did the simplest thing initially possible.

**Comment by grantjenks** [comment]
Fixed at 40ce0de
```
(env38) 2020-11-08 20:23:47 grantjenks @ GJ-MacBook-Air-19 : ~/repos/python-diskcache 
$ ls /tmp
com.apple.launchd.ZeqQUNvc1m	jupyter.lab.out
com.google.Keystone		powerlog
jupyter.lab.err
(env38) 2020-11-08 20:24:32 grantjenks @ GJ-MacBook-Air-19 : ~/repos/python-diskcache 
$ git log -1 --oneline
40ce0de (HEAD -> master) Use no hardcoded /tmp/diskcache/... paths in tests
(env38) 2020-11-08 20:24:43 grantjenks @ GJ-MacBook-Air-19 : ~/repos/python-diskcache 
$ pytest
================================== test session starts ==================================
platform darwin -- Python 3.8.4, pytest-6.0.1, py-1.9.0, pluggy-0.13.1
rootdir: /Users/grantjenks/repos/python-diskcache, configfile: tox.ini, testpaths: docs, diskcache, tests
plugins: xdist-2.0.0, django-3.9.0, forked-1.2.0, env-0.6.2, cov-2.10.1
gw0 [242] / gw1 [242]
................................................................................. [ 33%]
................................................................................. [ 66%]
................................................................................  [100%]
================================= 242 passed in 30.12s ==================================
(env38) 2020-11-08 20:25:18 grantjenks @ GJ-MacBook-Air-19 : ~/repos/python-diskcache 
$ ls /tmp
com.apple.launchd.ZeqQUNvc1m	jupyter.lab.out
com.google.Keystone		powerlog
jupyter.lab.err
```
